src/main/java/de/mediathekview/mserver/crawler/basic/M3U8Parser.java

```diff
@@ -0,0 +1,80 @@
+package de.mediathekview.mserver.crawler.basic;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Parser für M3U8-Dateien
+ */
+public class M3U8Parser {
+  public List<M3U8Dto> parse(String aM3U8Data) {
+    List<M3U8Dto> result = new ArrayList<>();
+    
+    List<String[]> pairs = getLinePairs(aM3U8Data);
+    for(String[] pair : pairs) {
+      M3U8Dto dto = new M3U8Dto(pair[1]);
+      parseMeta(pair[0], dto);
+      
+      result.add(dto);
+    }
+    
+    return result;
+  }
+  
+  private static void parseMeta(String aLine, M3U8Dto aDto) {
+    int index = aLine.indexOf(":");
+    if (index > 0) {
+      String line = aLine.substring(index + 1);
+      
+      while((index = line.indexOf(",")) > 0) {
+        // Sicherstellen, dass Index für , nicht innerhalb eines mit " begrenzten Wertes liegt
+        int indexQuote = line.indexOf("\"");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Put single-quotes around '\"' to use the faster "indexOf(char)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)